src/views/Editor.vue

```diff
@@ -775,7 +777,7 @@ export default {
 			try {
 				const { data } = await apiClient.get(`/datasets/${id}`)
 				this.$store.commit('resetMetadata')
-				this.$store.commit('setMetaxRecord', null)
+				this.$store.dispatch('clearMetaxRecord')
```

Comment:
I presume that this is async on purpose?

Reply:
It actually ran synchronously in this case (the action function is not async) but it seems that dispatch returns a Promise anyway. It feels dirty to assume that something that returns a Promise runs synchronously so I moved this stuff to a mutation.